lib/GitterUser.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+var Promise = require("bluebird");
+
+var rp = require("request-promise");
+
+// miliseconds of interval between presence=online updates
+var ONLINE_PERIOD = 4 * 60 * 1000;
+
+// miliseconds of grace period before users count as offline
+var OFFLINE_GRACE = 3 * 60 * 1000;
+
+function GitterUser(bridge, remoteUserModel) {
+  this._bridge = bridge;
+  this._remoteUserModel = remoteUserModel;
+
+  this._presenceRooms = {};
+  this._ispresent = false;
+  this._presenceinterval = null;
+  this._presencedelay = null;
+}
+
+GitterUser.prototype.gitterId = function() {
+  return this._remoteUserModel.getId();
+};
+
+GitterUser.prototype.gitterUsername = function() {
+  return this._remoteUserModel.get("username");
+};
+
+GitterUser.prototype.getIntent = function() {
+  return this._intent = this._intent || (
+      this._bridge.getIntentFromLocalpart('gitter_' + this.gitterUsername()));
+};
+
+GitterUser.prototype.update = function(user) {
+  return Promise.all([
+      this.updateDisplayName(user.displayName),
+      this.updateAvatar(user.avatarUrlMedium)
+  ]);
+};
+
+GitterUser.prototype.updateDisplayName = function(name) {
+  if (this._remoteUserModel.get("display_name") == name)
+    return Promise.resolve();
+
+  var intent = this.getIntent();
+
+  return intent.setDisplayName(name + ' (Gitter)').then(() => {
+    this._remoteUserModel.set("display_name", name);
+
+    return this._bridge.putRemoteUser(this._remoteUserModel);
+  });
+};
+
+GitterUser.prototype.updateAvatar = function(url) {
+  if (this._remoteUserModel.get("avatar_url") == url)
+    return Promise.resolve();
```

Comment:
As dbkr says, if you're going to indent this single statement (and `:45`), I insist you add braces to avoid [goto fail](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/) bugs.

I would personally force braces everywhere, even for entirely-single-line-if-statements, but I will tolerate that being the special snowflake exception.